src/Lorentz/Contracts/RegistryDAO.hs

```diff
@@ -0,0 +1,125 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | A Registry DAO can be used to store key value pair `(k, v)`
+module Lorentz.Contracts.RegistryDAO
+  ( registryDaoContract
+  , config
+  )
+  where
+
+import Lorentz
+
+import qualified Lorentz.Contracts.BaseDAO as DAO
+import Lorentz.Contracts.RegistryDAO.Types
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+import Universum hiding (drop, swap, (>>), some)
+import Util.Markdown
+
+-- | Proposal check: `s = size(pack(proposalMetadata))` `(a * s + b)`
+registryDaoProposalCheck
+  :: forall s k v.
+     ( NicePackedValue (RegistryDaoProposalMetadata k v)
+     )
+  => DAO.ProposeParams (RegistryDaoProposalMetadata k v) : s
+  :-> Bool : s
+registryDaoProposalCheck = do
+  let a :: Natural = 1; b :: Natural = 0
```

Comment:
This `a` and `b` parameters should be taken from the storage. They should be configurable on origination and modifiable in runtime by making a proposal. Also I would name they more descriptively, I just didn't bother coming up with better name, but it should be done.

Reply:
I've updated it so that the `a`, `b`, .. are taken from storage now. I also added `ConfigProposal` which will update this value when accepted. 

I've renamed these value to:
```
data RegistryDaoContractExtra k v = RegistryDaoContractExtra
  { ceRegistry :: BigMap k (RegistryEntry v)
  , ceFrozenScaleValue :: Natural -- a
  , ceFrozenExtraValue :: Natural -- b
  , ceSlashScaleValue :: Natural -- c
  , ceSlashDivisionValue :: Natural -- d
  }
```

Not sure about the naming, hope to get advice on this as well.

Next, I'll work on making the tests for this new config proposal.

Reply:
I've updated the `ConfigProposal` to match the spec now:
```
data ConfigProposal = ConfigProposal
  { cpFrozenScaleValue :: Maybe Natural -- a
  , cpFrozenExtraValue :: Maybe Natural -- b
  , cpSlashScaleValue :: Maybe Natural -- c
  , cpSlashDivisionValue :: Maybe Natural -- d
  , cpMaxProposalSize :: Maybe Natural -- s_max
  }
```

The only thing left is tests for this config proposal and some code cleanup.